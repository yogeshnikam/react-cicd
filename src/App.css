.app-container {
    min-height: 100vh;
    display: flex;
    flex-direction: column;
}

.nav-container {
    background-color: #333;
    padding: 1rem;
    display: flex;
    gap: 1rem;
}

.nav-link {
    color: white;
    text-decoration: none;
    padding: 0.5rem 1rem;
    border-radius: 4px;
    transition: background-color 0.3s;
}

.nav-link:hover {
    background-color: #555;
}

.main-content {
    flex: 1;
    padding: 2rem;
}

.not-found-container {
    text-align: center;
    padding: 2rem;
}

.home-link {
    display: inline-block;
    margin-top: 1rem;
    padding: 0.5rem 1rem;
    background-color: #007bff;
    color: white;
    text-decoration: none;
    border-radius: 4px;
    transition: background-color 0.3s;
}

.home-link:hover {
    background-color: #0056b3;
}

.loading-container {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    min-height: 200px;
}

.loading-spinner {
    width: 40px;
    height: 40px;
    border: 4px solid #f3f3f3;
    border-top: 4px solid #3498db;
    border-radius: 50%;
    animation: spin 1s linear infinite;
}

@keyframes spin {
    0% { transform: rotate(0deg); }
    100% { transform: rotate(360deg); }
} 