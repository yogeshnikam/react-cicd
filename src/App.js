import React, { Suspense, lazy } from 'react';
import { Routes, Route, Link } from 'react-router-dom';
import Preload from './components/Preload';
import './App.css';

// Lazy load components
const Home = lazy(() => import('./routes/Home'));
const About = lazy(() => import('./routes/About'));
const NotFound = lazy(() => import('./routes/NotFound'));

// Loading component
const Loading = () => (
  <div className="loading-container">
    <div className="loading-spinner"></div>
    <p>Loading...</p>
  </div>
);

const App = () => {
  return (
    <div className="app-container">
      <Preload />
      <nav className="nav-container">
        <Link to="/" className="nav-link">Home</Link>
        <Link to="/about" className="nav-link">About</Link>
      </nav>
      <main className="main-content">
        <Suspense fallback={<Loading />}>
          <Routes>
            <Route path="/" element={<Home />} />
            <Route path="/about" element={<About />} />
            <Route path="*" element={<NotFound />} />
          </Routes>
        </Suspense>
      </main>
    </div>
  );
};

export default App;