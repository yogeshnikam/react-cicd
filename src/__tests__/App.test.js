import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import App from '../App';

// Mock lazy-loaded components
jest.mock('../routes/Home', () => () => <div>Home Component</div>);
jest.mock('../routes/About', () => () => <div>About Component</div>);
jest.mock('../routes/NotFound', () => () => <div>Not Found Component</div>);
jest.mock('../components/Preload', () => () => <div data-testid="preload" />);

// Mock the store
const mockStore = configureStore([]);

describe('App Component', () => {
    let store;

    beforeEach(() => {
        store = mockStore({
            users: {
                users: [],
                loading: false,
                error: null
            }
        });
    });

    it('renders navigation links', () => {
        render(
            <Provider store={store}>
                <BrowserRouter>
                    <App />
                </BrowserRouter>
            </Provider>
        );

        expect(screen.getByText('Home')).toBeInTheDocument();
        expect(screen.getByText('About')).toBeInTheDocument();
    });

    it('renders Preload component', () => {
        render(
            <Provider store={store}>
                <BrowserRouter>
                    <App />
                </BrowserRouter>
            </Provider>
        );

        expect(screen.getByTestId('preload')).toBeInTheDocument();
    });

    it('matches snapshot', () => {
        const { container } = render(
            <Provider store={store}>
                <BrowserRouter>
                    <App />
                </BrowserRouter>
            </Provider>
        );

        expect(container).toMatchSnapshot();
    });
});
