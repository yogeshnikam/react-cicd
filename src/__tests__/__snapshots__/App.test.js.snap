// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component matches snapshot 1`] = `
<div>
  <div
    class="app-container"
  >
    <div
      data-testid="preload"
    />
    <nav
      class="nav-container"
    >
      <a
        class="nav-link"
        href="/"
      >
        Home
      </a>
      <a
        class="nav-link"
        href="/about"
      >
        About
      </a>
    </nav>
    <main
      class="main-content"
    >
      <div>
        About Component
      </div>
    </main>
  </div>
</div>
`;
